android-lib/src/androidTest/java/com/kinvey/androidTest/store/DeltaSetNewTest.java

```diff
@@ -0,0 +1,766 @@
+    }
+
+    @Test
+    public void testSyncWithQueryAfterUpdating() throws InterruptedException {
+        initDeltaSetCachedCollection(StoreType.SYNC);
+        testManager.save(store, new Person(TEST_USERNAME));
+        testManager.save(store, new Person(TEST_USERNAME));
+        testManager.save(store, new Person(TEST_USERNAME_2));
+        CustomKinveySyncCallback syncCallback  = testManager.sync(store, usernameQuery);
+        assertEquals(2, syncCallback.getResult().getCount());
+        updateItem(usernameQuery);
+        syncCallback  = testManager.sync(store, usernameQuery);
+        assertEquals(1, syncCallback.getResult().getCount());
+    }
+
+    @Test
+    public void testSyncAfterDisablingDeltaSet() throws InterruptedException {
+        initDeltaSetCachedCollection(StoreType.SYNC);
+        testManager.save(store, new Person(TEST_USERNAME));
+        testManager.push(store);
+        CustomKinveySyncCallback syncCallback = testManager.sync(store, emptyQuery);
+        assertEquals(1, syncCallback.getResult().getCount());
+        syncCallback = testManager.sync(store, emptyQuery);
+        assertEquals(0, syncCallback.getResult().getCount());
+        store.setDeltaSetCachingEnabled(false);
+        syncCallback = testManager.sync(store, emptyQuery);
+        assertEquals(1, syncCallback.getResult().getCount());
+        syncCallback = testManager.sync(store, emptyQuery);
+        assertEquals(1, syncCallback.getResult().getCount());
+    }
+
+    /* The test aims at confirming that find with forceNetwork works as intended */
+    @Test
+    public void testForceNetwork() throws InterruptedException {
+        DataStore<Person> networkStore = DataStore.collection(Person.DELTA_SET_COLLECTION, Person.class, StoreType.NETWORK, client);
+        networkStore.setDeltaSetCachingEnabled(true);
+        testManager.cleanBackend(networkStore, StoreType.NETWORK);
+        testManager.save(networkStore, new Person(TEST_USERNAME));
+        testManager.save(networkStore, new Person(TEST_USERNAME_2));
+        assertEquals(2, testManager.find(networkStore, emptyQuery).getResult().getResult().size());
+
+        List<Person> people = testManager.find(networkStore, emptyQuery).getResult().getResult();
+        Person person = people.get(0);
+        person.setAge("20");
+        testManager.save(networkStore, person);
+        assertEquals(2, testManager.find(networkStore, emptyQuery).getResult().getResult().size());
+
+        people = testManager.find(networkStore, emptyQuery).getResult().getResult();
+        person = people.get(0);
+        testManager.delete(networkStore, person.getId());
+        assertEquals(1, testManager.find(networkStore, emptyQuery).getResult().getResult().size());
+    }
+
+    @Test
+    public void testForceNetworkWithQuery() throws InterruptedException {
+        DataStore<Person> networkStore = DataStore.collection(Person.DELTA_SET_COLLECTION, Person.class, StoreType.NETWORK, client);
```

Comment:
I think the goal of this test is to use a `SYNC` store, but then override the read behavior with `ReadPolicy.FORCE_NETWORK`, to verify that it works as expected, not to just use a `NETWORK` store.

Reply:
Updated test.